### PR TITLE
Adding Github support

### PR DESCRIPTION
**Pull request by MovieStoreGuy**

### CODE REVIEW

File: providers/GoogleCloudPlatform/resource.go [hunk 1]
```diff
@@ -11,7 +14,7 @@ type userAccount struct {
 type userCollection []userAccount
 
 func (r userAccount) GetName() string {
-	return r.Name
+	return r.Email
```
Comment (MovieStoreGuy) [inline providers/GoogleCloudPlatform/resource.go]: This can be considered the unique name for the object

File: providers/GitHub/manager.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+func NewManager() (overwatch.IamPolicyManager, error) {
+	return &manager{
+		base:      abstract.DefaultManager(),
+		resources: map[string]map[string]overwatch.IamResource{},
+	}, nil
+}
+
+func (m *manager) LoadConfiguration(conf overwatch.IamManagerConfig) error {
```
Comment (AlexsJones) [inline providers/GitHub/manager.go]: Why is this not in the constructor?

File: providers/GitHub/manager.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	}
+	// Directory is made up of "path/<Provider>/<Organisation>/<ResourceType>/*.ya?ml"
+	dir := path.Join(m.base.Storer.GetPath(), "Github", m.organisation, "Repos")
+	loaded, err := abstract.ReadFiles(dir, projectTransformer)
+	if err != nil {
+		return err
+	}
+	for _, obj := range loaded {
+		if _, ok := m.resources[obj.GetType()]; !ok {
+			m.resources[obj.GetType()] = map[string]overwatch.IamResource{}
```
[the remark this answers is not in the document]
Comment (MovieStoreGuy) [reply providers/GitHub/manager.go]: It creates an empty map

File: providers/default/manager.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+func DefaultManager() *Manager {
+	return &Manager{
+		Expire: time.Now(),
+	}
+}
+
+func (m *Manager) Readconfig(conf overwatch.IamManagerConfig) error {
+	m.Conf = &conf
```
[the remark this answers is not in the document]
Comment (MovieStoreGuy) [reply providers/default/manager.go]: As the function passes in a copy of the original conf, I want to store a reference to it so further functions down don't create a new copy but modify the copy on the heap.

File: providers/default/manager.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+		// Only process files that we expect
+		if !regexp.MustCompile("^.*\\.(yaml|yml)$").MatchString(file.Name()) {
+			continue
+		}
+		filepath := path.Join(dir, file.Name())
+		buff, err := ioutil.ReadFile(filepath)
+		if err != nil {
+			return nil, err
+		}
+		collection = append(collection, transformer(buff)...)
```
[the remark this answers is not in the document]
Comment (MovieStoreGuy) [reply providers/default/manager.go]: Yeah, so the transformer function is meant to handle errors (by panicing) though, it wouldn't be hard to change it to the usual ```data, err``` return.

**Pull request merged**